insights/__init__.py

```diff
@@ -332,7 +332,7 @@ class Args(object):
         broker = dr.Broker()
         broker[ExecutionContext] = ctx
         for spec, content in specs.items():
-            broker[spec] = content if spec.multi_output else content[-1]
+            broker[spec] = content if hasattr(spec, 'multi_output') and spec.multi_output else content[-1]
```

Comment:
Instead of `hasattr` we can use `dr.DELEGATES[spec].multi_output`, which will get the decorator instance of the datasource instead of the datasource itself, and the datasource decorator should always have a `multi_output` attribute.

Reply:
That makes sense, changes have been made and tested.